ansible_base/activitystream/signals.py

```diff
@@ -7,6 +7,20 @@
 logger = logging.getLogger('ansible_base.activitystream.signals')
 
 
+def _always_exclude_fields():
+    """
+    Returns a list of 2-tuples of (model_or_base_class, 'field_name') that
+    should always be excluded from activity stream entries.
+    """
+
+    # The whole reason this is a function is to avoid circular imports.
+    from django.contrib.auth.models import AbstractUser
+
+    return [
+        (AbstractUser, 'last_login'),
```

Comment:
wouldn't it be easier to just add an ignore list on the `AbstractUser` model?

Reply:
We could, but I wasn't sure about modifying Django base classes, it felt wrong to me.

Reply:
As long as you give it a name that won't conflict with anything else, you should be fine. Of course, there is a distinction between the DAB AbstractUser and the Django AbstactUser. But I would prefer to do this DAB-specific, because it gives apps using this library the ability to change the behavior if they wish.

Going by history, there is likely to be a lot of iteration as QE looks at the activity stream entries and goes "that shouldn't be there!" You don't want to require a DAB patch, which is also why I think an ignore list on the model is a good idea.

Reply:
There's already a model-specific ignore list, models can set `activity_stream_excluded_field_names` and `activity_stream_limit_field_names` to exclude fields from showing up, or limit it to a subset of fields that can show up.

DAB doesn't have an `AbstractUser` concept. We could introduce one, but I think there was a conscious decision not to, a while back.

So I think the options are:

- Do it as-is in this PR so we can maintain a list of models -> fields to always exclude
- Do it on the Django `AbstractUser` model somewhere, by setting `AbstractUser.activity_stream_excluded_field_names = ['last_login']`
- Make every app set `activity_stream_excluded_field_names` in their own User model (but that doesn't work for something like AWX which doesn't use its own `User` model).

Reply:
> There's already a model-specific ignore list, models can set activity_stream_excluded_field_names and activity_stream_limit_field_names to exclude fields from showing up, or limit it to a subset of fields that can show up.

Maybe I'm just not getting it. Why wouldn't you use what you already developed and add "last_login" to `activity_stream_excluded_field_names`?

So I would say the option (3). I'm not worried about AWX. Yes, it uses `auth.User`, but it already adds _many_ things to the class in its models `__init__.py`. Take summary_fields for example. It's totally not pattern-breaking at all to glue that list onto an imported User model.

Reply:
> Why wouldn't you use what you already developed and add "last_login" to `activity_stream_excluded_field_names`?

Because you opened #285 :sweat_smile: 

If we want this to be something that the apps do to their own user model, then there's nothing to do here, except maybe exclude it in test_app's User model, right?

Reply:
> So I would say the option (3).

I updated this to just exclude the field in test_app's User model.

Reply:
It was filed from the experience of using test_app. Sometimes it is hard to separate that from other issues... but still an issue.

Reply:
It's all good :smile: It's nice if we don't have to maintain Yet Another List of hardcoded exceptions in DAB.